File: 1846.减小和重新排列数组后的最大元素/solution.cpp
class Solution {
public:
    int maximumElementAfterDecrementingAndRearranging(vector<int>& arr) {
        sort(arr.begin(),arr.end());
        int sz=arr.size();
        arr.at(0)=1;
        for (int i=1;i<sz;i++) {
            arr.at(i)=min(arr.at(i),arr.at(i-1)+1);
        }
        return arr.back();
    }
};

File: 1846.减小和重新排列数组后的最大元素/solution.go
func maximumElementAfterDecrementingAndRearranging(arr []int) int {
	sort.Ints(arr)
	arr[0]=1
	sz:=len(arr)
	for i:=1;i<sz;i++ {
		arr[i]=min(arr[i-1]+1,arr[i])
	}
	return arr[sz-1]
}
func min(a,b int) int {
	if a<b {
		return a
	}
	return b
}